### PR TITLE
Fix updating segments' NodeIds correctly after LoadBalance

### PR DESCRIPTION
**Pull request by yah01**
Signed-off-by: yah01 <yang.cen@zilliz.com>
#16705 #16644

### CODE REVIEW

File: internal/querycoord/task.go [hunk 1]
```diff
@@ -2300,6 +2306,26 @@ func (lbt *loadBalanceTask) globalPostExecute(ctx context.Context) error {
...
+
+			for _, segment := range segments {
+				wg.Add(1)
+				go func(segment *querypb.SegmentInfo) {
+					defer wg.Done()
+
+					segment.NodeID = -1
+					segment.NodeIds = removeFromSlice(segment.NodeIds, lbt.SourceNodeIDs...)
+					if len(segment.NodeIds) > 0 {
+						segment.NodeID = segment.NodeIds[0]
```
Comment (sunby) [inline internal/querycoord/task.go:2318]: Why you set the first element in NodeIds as NodeID?

Comment (yah01) [reply internal/querycoord/task.go:2318]: The NodeID may refer to an offline node, so set it to an online node here

**Review by congqixia — APPROVED** [review]
/lgtm

File: internal/querycoord/task.go (hunk 1 above)
Comment (congqixia) [reply internal/querycoord/task.go:2318]: @yah01 @sunby is this comment resolved?

**Comment by yah01** [comment]
#16881

**Pull request merged**